File: Backend/src/main/java/com/example/backend/dto/ArticleResponseDto.java
package com.example.backend.dto;

import java.time.LocalDateTime;
import java.util.List;

public record ArticleResponseDto(
        Integer id,
        String title,
        ArticleCategoryResponseDto category,
        List<String> mainWords,
        String author,
        String imageUrl,
        LocalDateTime createdAt,
        LocalDateTime updatedAt,
        Integer likesCount,
        Boolean isFavorite,
        Boolean isLiked
) {
}

File: MyApplication/data/src/main/java/com/example/data/Hints/datasource/remote/dto/ArticleResponseDto.kt
package com.example.data.Hints.datasource.remote.dto

data class ArticleResponseDto(
    val id: Int,
    val title: String,
    val category: ArticleCategoryDto,
    val mainWords: List<String>?,
    val author: String?,
    val imageUrl: String?,
    val createdAt: String,
    val updatedAt: String,
    val likesCount: Int,
    val isFavorite: Boolean,
    val isLiked: Boolean
)
